packages/@lwc/engine-core/src/framework/component.ts

```diff
@@ -24,13 +25,16 @@ const signedTemplateMap: Map<LightningElementConstructor, Template> = new Map();
 export function registerComponent(
     // We typically expect a LightningElementConstructor, but technically you can call this with anything
     Ctor: any,
-    { tmpl }: { tmpl: Template }
+    { tmpl, sel }: { tmpl: Template; sel: string }
 ): any {
     if (isFunction(Ctor)) {
         if (process.env.NODE_ENV !== 'production') {
             checkVersionMismatch(Ctor, 'component');
         }
         signedTemplateMap.set(Ctor, tmpl);
+        if (isComponentConstructor(Ctor)) {
+            Ctor.sel = sel;
+        }
```

Comment:
In this POC `registerComponent` would store the tag name on the Ctor as an example.